cloudify_cli/utils.py

```diff
@@ -122,17 +122,53 @@ def inputs_to_dict(resource, resource_name):
     if not resource:
         return None
     try:
-        if os.path.exists(resource):
-            with open(resource, 'r') as f:
-                return yaml.load(f.read())
-        else:
-            return yaml.load(resource)
-    except yaml.error.YAMLError as e:
-        msg = ("'{0}' must be a valid YAML. {1}"
-               .format(resource_name, str(e)))
+        # parse resource as string representation of a dictionary
+        parsed_dict = plain_string_to_dict(resource)
+    except CloudifyCliError:
+        try:
+            # if resource is a path - parse as a yaml file
+            if os.path.exists(resource):
+                with open(resource, 'r') as f:
+                    return yaml.load(f.read())
+            else:
+                # parse resource content as yaml
+                parsed_dict = yaml.load(resource)
+        except yaml.error.YAMLError as e:
+            msg = ("'{0}' is not a valid YAML. {1}"
+                   .format(resource_name, str(e)))
+            raise CloudifyCliError(msg)
+
+    if isinstance(parsed_dict, dict):
+        return parsed_dict
+    else:
+        msg = "Invalid input: {0}. {1} must be either be a path to a valid " \
```

Comment:
I think the message here is a bit misleading.
If the key=value syntax is applied, you are guaranteed to get a valid dictionary.
If on the other hand I supply a YAML string or YAML file that don't represent a dict, this is where the trouble lies.
So the message should be clear that the YAML variations should represent a valid dict/hash/map/object